modules/ex/test/exoscale/ex/test/core_test.clj

```diff
@@ -128,9 +129,19 @@
                        (throw (ex-info "boom" {:exoscale.ex/type ::bar}))
                        (catch ::bar e# 1)))))))
 
-(deftest test-thrown-ex-data
+(deftest test-thrown-ex-type
   (is (thrown-ex-info-type? ::foo (throw (ex/ex-info "bar" ::foo)))))
 
+(deftest test-thrown-ex-msg
+  (is (thrown-with-ex-info-msg? ::foo #"bar" (throw (ex/ex-info "bar" ::foo)))))
+
+(deftest test-thrown-ex-cause
+  (let [x (ex/ex-info "bar"
+                      ::foo
+                      {:a 1
+                       :message "the-cause"})]
```

Comment:
Ok, now I get what you are actually trying to achieve, sorry I was slow on that one. 

You want to test with a regex a value in the **ex-data** (:message). 

If that's the goal I think we might want to make the key a parameter:


`(thrown-with-ex-info-msg-at-key? ::foo :message #"bar" (ex/ex-info "foo" ::foo {:message "bar"})))`

Or maybe it's even better to make it an arbitrary predicate, so you can match non-string things.

`(thrown-with-ex-info-pred? ::foo #(re-find #"bar" (:message %) ) (ex/ex-info "foo" ::foo {:message "bar"})))`

or both.